### ISSUE TITLE
Ionic CLI - ionic capacitor run android

### ISSUE THREAD

**Issue by antoinecfmws** [ionic-team/ionic-cli#4057]
Hello,

One of the ionic cli commands needs a bit of tweaking. `ionic capacitor run android -l` runs by default `ng run app:serve --host=localhost --port=8100` with the server being localhost and port 8100, then it does a `capacitor copy android` and opens the android studio.

When we launch the android emulator, it is watching for changes on localhost:8100, however localhost:8100 is not reachable from the device emulator, the host machine is at 10.0.2.2 (https://developer.android.com/studio/run/emulator-networking).

It seems like we either need a rule to forward localhost to 10.0.2.2 or to let the android studio / emulator handles their own stuff, we can offer 2 urls in capacitor.config.json one url for the angular serve (localhost:8100) and one for the android emulator (10.0.2.2:8100).

Thoughts?

**Comment by dwieeb** [comment]
@jcesarmobile Thoughts here? 

The `--public-host` option may be useful: https://github.com/ionic-team/ionic-cli/issues/3558 (to be implemented)

**Comment by jcesarmobile** [comment]
I think it should set the url to the local IP of the computer doing serve, that way it will work in all iOS and android devices (if connected to the same network) and emulators/simulators, while 10.0.2.2 will only work on the android emulator

**Comment by dwieeb** [comment]
So that would mean `ionic cap run -l` would always require `--address=0.0.0.0`.

**Comment by funky-jojo** [comment]
I agree something needs to be fixed.  I'm just getting started with ionic+android development and been struggling with this...

**Comment by dwieeb** [comment]
AFAIK there isn't a magic way to make live reload work for everybody every time. It's tricky, and the dev needs to understand what's going on so they don't run into obscure networking errors and get totally stuck.

We do, however, include the `--external` option in all the examples of live reload with Capacitor. This is an opt-in flag so hopefully the dev tries to understand what it is and what it's doing. I wrote a more in-depth [explanation of live reload in our docs](https://ionicframework.com/docs/cli/livereload), but I need to update it for Capacitor.

**Comment by funky-jojo** [comment]
With the --external option, the ionic cli runs the app with host=0.0.0.0 , and the dev server ends up running as:


    Local: http://localhost:8100

       External: http://10.0.0.59:8100

But the end result is the same – the web view in Android emulator tries connecting to localhost instead of 10.0.0.59.  Not sure where/how to change this.


________________________________
